api/src/main/java/org/openmrs/module/htmlformentryui/element/ObsFromFragmentElement.java

```diff
@@ -0,0 +1,328 @@
+			throw new IllegalArgumentException("Parameter initFragmentParamName cannot be blank");
+		}
+		StringBuilder fragmentUrlBuilder = new StringBuilder();
+		fragmentUrlBuilder.append(fragment);
+		fragmentUrlBuilder.append("?");
+		fragmentUrlBuilder.append(parameters.get("fragmentParams"));
+		try {
+			fragmentParams = UiIncludeTagHandler.paramsToMap(fragmentUrlBuilder.toString());
+							
+		} catch (URISyntaxException e) {
+			throw new FragmentException("Invalid fragment URI: " + fragmentUrlBuilder.toString() , e);
+		}
+	}
+	
+	@Override
+	public void handleSubmission(FormEntrySession session, HttpServletRequest request) {
+		String formFieldName = (String) fragmentParams.get("formFieldName");
+		if (formFieldName != null) {
+			String valueText = request.getParameter(formFieldName);
+			if (session.getContext().getMode().equals(FormEntryContext.Mode.EDIT)) {
+				List<Obs> existingObservations = session.getContext().removeExistingObs(concept);
+				if (existingObservations != null) {
+					if (!existingObservations.isEmpty()) {
+						// get the first Obs 
+						session.getSubmissionActions().modifyObs(existingObservations.get(0), concept, convertToType(valueText), new Date(), null);
+						return;
+					}
+				}
+				
+			}
+			session.getSubmissionActions().createObs(concept, convertToType(valueText), new Date(), null);
+		}
+	}
+
+	@Override
+	public Collection<FormSubmissionError> validateSubmission(FormEntryContext context, HttpServletRequest request) {
+		return Collections.emptyList();
+	}
+
+	@Override
+	public String generateHtml(FormEntryContext context) {
+		initializeFragment(context);
+		if (FormEntryContext.Mode.VIEW == context.getMode()) {
+			StringBuilder sb = new StringBuilder();
+			sb.append("<p><b>");
+			sb.append(fragmentParams.get("label"));
+			sb.append("</b> : <span>");
+			sb.append(fragmentParams.get(initFragmentParamName));
+			sb.append("</span></p>");
+			return sb.toString();
+		}
+		try {
+			return uiUtils.includeFragment(viewProvider, fragment, fragmentParams);
+		} catch (NullPointerException e) {
+			// if we are validating/submitting the form, then this method is being called from a fragment action method
+            // and the UiUtils we have access to doesn't have a FragmentIncluder. That's okay, because we don't actually
+            // need to generate the HTML, so we can pass on this exception.
+            // (This is hacky, but I don't see a better way to do it.)
+            return "Submitting the form, so we don't generate HTML";
+		} catch (PageAction e) {
+			throw new IllegalStateException("Tried to include a fragment that threw a PageAction.", e);
+		}
+	}
+	
+	protected Object convertToType(String val) {
+		ConceptDatatype dataType = concept.getDatatype();
+		if (dataType.isDate()) {
+			return HtmlFormEntryUtil.convertToType(val, Date.class);
+		}
+		if (dataType.isDateTime() || dataType.isTime()) {
+			try {
+				DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
I would rather use these two constants https://github.com/openmrs/openmrs-module-uiframework/blob/master/api/src/main/java/org/openmrs/ui/framework/WebConstants.java#L9-L11 instead of hitting to the DB for such a simple use-case :)

Reply:
@samuelmale the date-time format should be consistent across the whole Ref App and is configured through that GP so that it can be changed at runtime and controlled globally. This is why there is no avoiding hitting the backend.

Does this show a performance hit?

Reply:
Ooohh makes sense to me :)